integreat_cms/cms/utils/translation_utils.py

```diff
@@ -46,3 +49,55 @@ def translate_link(message, attributes):
         link_text,
         after,
     )
+
+
+def mt_is_permitted(region, user, content_type, target_lang_slug):
+    """
+    Checks if a machine translation is permitted, i.e. if for the
+    given region, MT of the given content type is allowed and
+    MT into the target language is enabled for the requesting user.
+
+    :param region: the region attempting to machine translate
+    :type region: ~integreat_cms.cms.models.regions.region.Region
+
+    :param user: the user requesting the translation
+    :type user: ~django.contrib.auth.models.User
+
+    :param content_type: type of content which would be translated
+    :type content_type: ~integreat_cms.cms.models.abstract_content_model.AbstractContentModel
+
+    :param target_lang_slug: slug of the MT target language
+    :type target_lang_slug: str
+
+    :return: if the translation is permitted
+    :rtype: bool
+    """
+    permission_settings = {
+        "events": (region.machine_translate_events, "cms.change_event"),
+        "pages": (region.machine_translate_pages, "cms.change_page"),
+        "pois": (region.machine_translate_pois, "cms.change_poi"),
+    }
+
+    translation_permission_setting, content_permission_required = permission_settings[
+        content_type
+    ]
+
+    if translation_permission_setting == mtp.NO_ONE:
+        return False
+    # use "manage_translations" permission to determine if user has at least role Management
+    if translation_permission_setting == mtp.MANAGERS and not user.has_perm(
+        "manage_translations"
+    ):
+        return False
+    # machine translations are allowed, check if the user has permission to edit the content type
+    if not user.has_perm(content_permission_required):
+        return False
+
+    # else: user is permitted to edit and translate this content;
+    # determine if translation into language is permitted
+    target_lang_allowed = get_object_or_404(
+        region.language_tree_nodes,
+        language__slug=target_lang_slug,
+    ).machine_translation_enabled
+
+    return target_lang_allowed
```

Comment:
```suggestion
    mt_perms_setting, required_perm = permission_settings[content_type]

    if mt_perms_setting == mt_perms.NO_ONE:
        logger.debug(
            "Machine translations are disabled for content type %r in %r.",
            content_type,
            region,
        )
        return False

    mt_perm = "cms.manage_translations"
    if mt_perms_setting == mt_perms.MANAGERS and not user.has_perm(mt_perm):
        logger.debug(
            "Machine translations are only enabled for content type %r in %r for users with the permission %r.",
            content_type,
            region,
            mt_perm,
        )
        return False

    if not user.has_perm(required_perm):
        logger.debug(
            "Machine translations are only enabled for content type %r in %r for users with the permission %r.",
            content_type,
            region,
            required_perm,
        )
        return False

    target_lang = region.language_node_by_slug[target_lang_slug]

    if target_lang == region.default_language:
        logger.debug(
            "Machine translations are disabled for the default language %r in %r.",
            target_lang,
            region,
        )
        return False

    if not target_lang.machine_translation_enabled:
        logger.debug(
            "Machine translations are disabled for %r in %r.", target_lang, region
        )
        return False

    return True
```
During testing, I noticed that the `manage_translations` permission was not correctly checked, because the app label `cms.` was missing...
I took me a while to figure this out, so I suggest to add a bit of debug logging to make it a bit easier to follow the reason why the function returns `False`. This should also make the comment obsolete.

Also, querying the language via `region.language_node_by_slug` instead of `get_object_or_404` prevents an additional query because the language tree nodes are already prefetched for the region model anyways.

Apart from that, I think we could also check whether a language is the default language here? Then we don't have to check it each template...

Reply:
Oh no, sorry to hear that 😅  And thank you for investing the time to add the debug logic already!

Good idea about checking for the default language.